### PR TITLE
Add stop_random_instance

### PR DESCRIPTION
**Pull request by nazarenof**
Add subsequent tests for filter and oci client.
Reorganise the code to about DRY code.
Add a new common file with common functions

Signed-off-by: Nazareno Feito <n@nazarenofeito.org>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,109 +0,0 @@
-# Byte-compiled / optimized / DLL files
```
Comment (Lawouach) [inline .gitignore]: Any reason this file is deleted?

File: chaosoci/compute/actions.py [hunk 2]
```diff
@@ -1,26 +1,66 @@
 # -*- coding: utf-8 -*-
+from random import choice
+from typing import Any, List, Dict
+
 from oci.core import ComputeClient
+from oci.config import from_file
```
Comment (Lawouach) [inline chaosoci/compute/actions.py]: I like sorting the imports if I can, so:

```
from oci.config import from_file
from oci.core import ComputeClient
```

Maybe we will enable isort some day to help :)

File: chaosoci/compute/actions.py [hunk 3]
```diff
@@ -1,26 +1,66 @@
...
 
 from chaoslib.types import Configuration, Secrets
+from chaoslib.exceptions import ActivityFailed
```
Comment (Lawouach) [inline chaosoci/compute/actions.py]: same comment

File: chaosoci/compute/actions.py [hunk 4]
```diff
@@ -1,26 +1,66 @@
...
+
+    return ret
+
+
+def stop_random_instance(filters: List[Dict[str, Any]],
+                         compartment_id: str = None,
+                         force: bool = False,
+                         configuration: Configuration = None,
+                         secrets: Secrets = None) -> OCIResponse:
+    """ Stop a a random compute instance within a given compartment.
```
Comment (Lawouach) [inline chaosoci/compute/actions.py]: I prefer if the docstring starts on a new line:

```
"""
Stop a ...
```

File: chaosoci/compute/common.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+from oci.core import ComputeClient
+
+from chaoslib.exceptions import ActivityFailed
+
+__all__ = ["get_instances", "filter_instances"]
+
+
+def get_instances(client: ComputeClient = None,
+                  compartment_id: str = None) -> List:
```
Comment (Lawouach) [inline chaosoci/compute/common.py]: Returns a List of ?

File: chaosoci/compute/common.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    while instances_raw.has_next_page:
+        instances_raw = client.list_instances(compartment_id=compartment_id,
+                                              page=instances_raw.next_page)
+        instances.extend(instances_raw.data)
+
+    return instances
+
+
+def filter_instances(instances: List = None,
+                     filters: Dict[str, Any] = None) -> List:
```
Comment (Lawouach) [inline chaosoci/compute/common.py]: same question :)

File: .gitignore (hunk 1 above)
Comment (nazarenof) [reply .gitignore]: It was not meant to do that, good catch. I have restored it.
Will be there on the updated PR.

File: chaosoci/compute/actions.py (hunk 2 above)
Comment (nazarenof) [reply chaosoci/compute/actions.py]: I've installed isort, I will sort them all, I was thinking about this a few days back.

File: chaosoci/compute/actions.py (hunk 3 above)
Comment (nazarenof) [reply chaosoci/compute/actions.py]: ditto.

File: chaosoci/compute/actions.py (hunk 4 above)
Comment (nazarenof) [reply chaosoci/compute/actions.py]: Well, still agrees with PEP 257, so I'm not bother in accepting the project's style.

File: chaosoci/compute/common.py (hunk 5 above)
Comment (nazarenof) [reply chaosoci/compute/common.py]: Would this impact something I am not aware of? or just in terms of improving annotations?

File: chaosoci/compute/common.py (hunk 6 above)
Comment (nazarenof) [reply chaosoci/compute/common.py]: Thought about it better; it is better to know precisely what we are returning; fixed.

**Comment by nazarenof** [comment]
We are ready to go now; thank you.

File: chaosoci/compute/common.py (hunk 5 above)
Comment (Lawouach) [reply chaosoci/compute/common.py]: The latter. It's more documentation and helping editors.

Comment (Lawouach) [reply chaosoci/compute/common.py]: We may also enable mypy someday.

**Pull request merged**